exercises/react/canonical-data.json

```diff
@@ -46,313 +46,340 @@
-          "name": "output",
-          "type": "compute",
-          "inputs": ["input"],
-          "compute_function": "inputs[0] + 1"
-        }
-      ],
-      "operations": [
-        {
-          "type": "set_value",
-          "cell": "input",
-          "value": 3
-        },
-        {
-          "type": "expect_cell_value",
-          "cell": "output",
-          "value": 4
-        }
-      ]
+      "input": {
+        "cells": [
+          {
+            "name": "input",
+            "type": "input",
+            "initial_value": 1
+          },
+          {
+            "name": "output",
+            "type": "compute",
+            "inputs": ["input"],
+            "compute_function": "inputs[0] + 1"
+          }
+        ],
+        "operations": [
+          {
+            "type": "set_value",
+            "cell": "input",
+            "value": 3
+          },
+          {
+            "type": "expect_cell_value",
+            "cell": "output",
+            "value": 4
+          }
+        ]
+      },
+      "expected": {}
     },
     {
       "description": "compute cells can depend on other compute cells",
       "property": "react",
-      "cells": [
-        {
-          "name": "input",
-          "type": "input",
-          "initial_value": 1
-        },
-        {
-          "name": "times_two",
-          "type": "compute",
-          "inputs": ["input"],
-          "compute_function": "inputs[0] * 2"
-        },
-        {
-          "name": "times_thirty",
-          "type": "compute",
-          "inputs": ["input"],
-          "compute_function": "inputs[0] * 30"
-        },
-        {
-          "name": "output",
-          "type": "compute",
-          "inputs": ["times_two", "times_thirty"],
-          "compute_function": "inputs[0] + inputs[1]"
-        }
-      ],
-      "operations": [
-        {
-          "type": "expect_cell_value",
-          "cell": "output",
-          "value": 32
-        },
-        {
-          "type": "set_value",
-          "cell": "input",
-          "value": 3
-        },
-        {
-          "type": "expect_cell_value",
-          "cell": "output",
-          "value": 96
-        }
-      ]
+      "input": {
+        "cells": [
+          {
+            "name": "input",
+            "type": "input",
+            "initial_value": 1
+          },
+          {
+            "name": "times_two",
+            "type": "compute",
+            "inputs": ["input"],
+            "compute_function": "inputs[0] * 2"
+          },
+          {
+            "name": "times_thirty",
+            "type": "compute",
+            "inputs": ["input"],
+            "compute_function": "inputs[0] * 30"
+          },
+          {
+            "name": "output",
+            "type": "compute",
+            "inputs": ["times_two", "times_thirty"],
+            "compute_function": "inputs[0] + inputs[1]"
+          }
+        ],
+        "operations": [
+          {
+            "type": "expect_cell_value",
```

Comment:
Note that this `expect_cell_value` comes **before** a `set_value`, therefore it is not possible to **consistently** place `expect_cell_value` in `expected`. Ordering is important. To place `expect_cell_value` in `expected`, we would need to redefine how `expect_cell_value` works so that it may refer to a point in time before/after certain other `operations`.